src/test/java/org/javarosa/core/model/instance/geojson/GeoJsonExternalInstanceTest.java

```diff
@@ -110,6 +110,16 @@ public void parse_prioritizesTopLevelId() throws IOException {
         assertThat(featureCollection.getChildAt(0).getChild("id", 0).getValue().getValue(), is("top-level-id"));
     }
 
+    @Test
+    public void parse_allowsIntegerId() throws IOException {
+        TreeElement featureCollection = GeoJsonExternalInstance.parse("id", r("feature-collection-integer-id.geojson").toString());
+
+        assertThat(featureCollection.getChildAt(0).getNumChildren(), is(4));
+        assertThat(featureCollection.getChildAt(0).getChild("id", 0).getValue().getValue(), is("77"));
+
+        assertThat(featureCollection.getChildAt(1).getChild("id", 0).getValue().getValue(), is("78"));
```

Comment:
Unless I'm missing something, there is no reason to have more than one feature in the JSON to check that integer IDs get parsed correctly.

Reply:
This is not my best habit. While I like targeted tests, I'm also very paranoid about the 1-element case being special. You're right that it's really not necessary here and I've removed it.